models/cam-fv/shell_scripts/synth_obs_locs_to_seqs.csh

```diff
@@ -0,0 +1,154 @@
+#!/bin/tcsh
+#
+# DART software - Copyright UCAR. This open source software is provided
+# by UCAR, "as is", without charge, subject to all terms of use at
+# https://www2.cisl.ucar.edu/software/dart/download
+#
+# DART $Id$
+
+# ==============================================================================
+# Call create_obs_sequence to convert a text file of obs locations and types
+# into a set_def.out file.  
+# Call create_fixed_network_seq to create separate files for each time period. 
+# This one makes multiple files/day (2 or 4 are the most common), 
+# a single time per file.
+
+# (Copy to and) Submit from a case_dir, where there's 
+# + a text file of locations and types for input to create_obs_sequence,
+#   (The text file CAN NOT request identity obs unless 
+#    suitable cam{input,_phis}.nc files are provided.)
+# + in input.nml which was used to build the create_* programs.
+#   (minimize printed output by listing only the relevant obs types 
+#    in assimilate_these_obs_types, or utilities_nml:write_nml = 'none').
+# Caminput.nc and cam_phis.nc will be linked from the executable directory
+# where create_* programs were created.
+# The resolution is irrelevant because no identity obs will be requested.
+# The time period must be within a calendar month.
+# Edit the values below to change the dates and intervals.
+
+# ==============================================================================
+#PBS  -N synth_obs_locs_to_seqs.csh
+#PBS  -A P86850054
+#PBS  -q casper
+#PBS  -l select=1:ncpus=1:mpiprocs=1
+#PBS  -l walltime=00:20:00
+#PBS  -o synth_obs_locs_to_seqs_2017-12.out
+#PBS  -k eod
+#PBS  -j oe 
+
+#----------------------------------------------------------------------------
+#BSUB  -J synth_obs_locs_to_seqs_2017-12
+#BSUB  -n 1 
+#BSUB  -R "span[ptile=1]"
+#BSUB  -q shared_node_queue_for_this_setup_script
+#BSUB  -P your_account_there
+#BSUB  -W 2:00
+#BSUB  -u you@email.org
+#BSUB  -N  
+#BSUB  -a poe 
+#BSUB  -o synth_obs_locs_to_seqs_2017-12.out
+#BSUB  -e synth_obs_locs_to_seqs_2017-12.out
+# ==============================================================================
```

Comment:
When generating a month of obs_seq.in files it required ~3.5 minutes.
Since it's a single processor job, there's a reasonable chance I'd get away with running it interactively,
but it's so annoying when all of my cheyenne windows disappear if one job is judged to be abusive.

Reply:
To say I am shocked that it takes 3.5 minutes is the understatement of the century. How many observations are there? Are you writing ASCII obs ?   I'm wondering if simply using 'sed' to change the known time from the set_def.out to the desired time might not be 50x faster?

Reply:
There are 48,384 obs total.   The script runs create_obs_sequence
and then create_fixed_network sequence O(120) times to finish a month.
I wrote ascii files so that I could easily see if they look right.
If all that needs to change is the date, then sed would probably be a better choice.
I was following the precedents and the documentation rather than looking to refactor anything.

Reply:
before you change any code, it would be worth looking at your compile options in mkmf.template
If you are not buffering IO, writing obs_sequence files can be a bit slow because they are lots of small writes to the file.

Reply:
i think ascii output is the right choice for these files, but helen is right about buffering i/o to speed up writing obs_seq files.

i agree with tim that the batch stuff is encouraging people to think this is a slow process.   making them look for a super to run this seems misleading.  if it was me i'd leave the script as-is but remove the batch sections and add a comment that this can be run on a laptop or a single processor of a larger system and let the users decide what to do.

for cheyenne you should get an interactive batch node to run this without fear of getting kicked off the login nodes.  (that's good advice for anything you want to run interactively, actually.)

if you feel strongly about leaving the batch stuff there at least add a comment before that section about this being runnable on a laptop so users aren't mislead.

Reply:
My mkmf.template uses 
FFLAGS = -O2 -xHost -ipo -vec-report0 -assume buffered_io

Thanks for the reminder about the "interactive batch" (qcmd) mode on cheyenne.
I developed this on casper but was dismayed to find that there's no qcmd there (anymore?).
So I added batch commands instead of switching machines and executables.

There is the question of whether we should change any code now at all.
At Tuesday's standup Jeff asked us to wind this up immediately.
We could make some notes about future improvements and move on.
It's not used much.

I'll add a line about the likelihood that it doesn't need to be a batch job.

Reply:
I changed the acquisition of files to fail if the first choices are not found.
I added a comment about running the script interactively.